frontend/src/app/core/services/video.service.ts

```diff
@@ -0,0 +1,23 @@
+import { HttpClient, HttpParams } from "@angular/common/http";
+import { Injectable } from "@angular/core";
+import { Observable, Subject } from "rxjs";
+import { environment } from "../../../environments/environment";
+import { VideoListDto } from "../models/video-list-dto";
+
+@Injectable({
+  providedIn: 'root'
+})
+export class VideoService {
+  private authenticationStateChangeSubject = new Subject<boolean>();
+  public authenticationStateChanged = this.authenticationStateChangeSubject.asObservable();
+
+  private readonly videoPageWebAPIUrl: string = `${environment.webApiUrl}`;
+
+  constructor(private httpClient: HttpClient) {}
+
+  getUserVideos(id: string): Observable<VideoListDto> {
+    let params = new HttpParams().set('id', id);
+    
+    return this.httpClient.get<VideoListDto>(`${this.videoPageWebAPIUrl}/user/videos`, { params: params });
```

Comment:
shouldn't we include authentication token? also is it your responsibility to add logic on be regarding fetching our videos vs fetching videos of someone else?

Reply:
right now it's noones responsibility